app/src/main/java/org/walleth/TyleSdk.java

```diff
@@ -59,23 +59,21 @@ public void setCurrentBalanceObserver(LifecycleOwner owner, View v) {
                         currentTokenProvider.getCurrentToken()));
     }
 
+    private void transactionEntitiesRefresh(List<TransactionEntity> transactionEntities) {
+        if (transactionEntities != null) {
+            refresh();
+        }
+    }
+
     public void installTransactionObservers(LifecycleOwner owner) {
         if (currentAddressProvider.getValue() == null) {
             Address currentAddress = currentAddressProvider.getCurrent();
             ChainDefinition currentChain = networkDefinitionProvider.getCurrent().getChain();
             incomingTransactionsForAddress = sdkApp.getAppDatabase().getTransactions().getIncomingTransactionsForAddressOnChainOrdered(currentAddress, currentChain);
             outgoingTransactionsForAddress = sdkApp.getAppDatabase().getTransactions().getOutgoingTransactionsForAddressOnChainOrdered(currentAddress, currentChain);
 
-            incomingTransactionsForAddress.observe(owner, transactionEntities -> {
-                if (transactionEntities != null) {
-                    refresh();
-                }
-            });
-            outgoingTransactionsForAddress.observe(owner, transactionEntities -> {
-                if (transactionEntities != null) {
-                    refresh();
-                }
-            });
+            incomingTransactionsForAddress.observe(owner, this::transactionEntitiesRefresh);
+            outgoingTransactionsForAddress.observe(owner, this::transactionEntitiesRefresh);
         }
```

Comment:
Code duplication from when I made it a lambda.  Thought it was a good idea to encapsulate behavior.